python/tests/xir/test_program.py

```diff
@@ -35,50 +26,40 @@ def test_includes(self):
     # Test declarations
     #####################
 
-    @pytest.mark.parametrize("name", ["rx", "CNOT", "a_gate"])
-    @pytest.mark.parametrize("num_params", [0, 1, 42])
-    @pytest.mark.parametrize("num_wires", [1, 42])
-    def test_gate_declaration(self, name, num_params, num_wires):
-        """Test serializing gate declarations"""
-        decl = GateDeclaration(name, num_params=num_params, num_wires=num_wires)
+    @pytest.mark.parametrize("params", [["a", "b"], [], ["theta"]])
+    @pytest.mark.parametrize("wires", [(0,), (0, 2, 1), ("a", "b", "c")])
+    @pytest.mark.parametrize("declaration_type", ["gate", "operator", "output"])
+    def test_declarations(self, params, wires, declaration_type):
+        """Test serializing gate, operation and output declarations"""
+        decl = Declaration("name", params, wires, declaration_type=declaration_type)
 
         irprog = XIRProgram()
-        irprog._declarations["gate"].append(decl)
+        irprog._declarations[declaration_type].append(decl)
         res = irprog.serialize()
-        assert res == f"gate {name}, {num_params}, {num_wires};"
 
-    @pytest.mark.parametrize("name", ["sin", "COS", "arc_tan"])
-    @pytest.mark.parametrize("num_params", [0, 1, 42])
-    def test_func_declaration(self, name, num_params):
-        """Test serializing function declarations"""
-        decl = FuncDeclaration(name, num_params=num_params)
+        if len(params) == 0:
+            params_str = ""
+        else:
+            params_str = "(" + ", ".join(params) + ")"
 
-        irprog = XIRProgram()
-        irprog._declarations["func"].append(decl)
-        res = irprog.serialize()
-        assert res == f"func {name}, {num_params};"
+        wires_str = ", ".join(str(w) for w in wires)
+        assert res == f"{declaration_type} name{params_str}[{wires_str}];"
```

Comment:
Just a thought: if we make `wires` a list instead of a tuple then we can write
```suggestion
        assert res == f"{declaration_type} name{params_str}{wires};"
```

Reply:
Good point. I guess there might not actually be any real reason for using tuples for wires (and having them as lists might simplify things a bit). Although in this case this wouldn't work for the same reason as described [above](https://github.com/XanaduAI/jet/pull/61#discussion_r688010482).